libraries/resource_providers/api500/c7000/storage_pool_provider.rb

```diff
@@ -0,0 +1,80 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative 'storage_system_provider'
+
+module OneviewCookbook
+  module API500
+    module C7000
+      # StoragePool API500 C7000 provider
+      class StoragePoolProvider < API300::C7000::StoragePoolProvider
+        include OneviewCookbook::RefreshActions::RequestRefresh
+
+        def load_storage_system
+          raise("Unspecified property: 'storage_system'. Please set it before attempting this action.") unless @new_resource.storage_system
+          data = {
+            hostname: @new_resource.storage_system,
+            name: @new_resource.storage_system
+          }
+          @item.set_storage_system(load_resource(:StorageSystem, data))
+        end
+
+        def update_manage_state(managed)
+          status = 'managed'
+          status = 'unmanaged' unless managed
+          return Chef::Log.info("#{@resource_name} '#{@name}' is already #{status}") if @item['isManaged'] == managed
+          @context.converge_by "#{@resource_name} '#{@name}' is now #{status}" do
+            @item.update(isManaged: managed)
```

Comment:
Suggestion: In the SDK, a method was implemented ```manage(Boolean)``` [here](https://github.com/HewlettPackard/oneview-sdk-ruby/blob/master/lib/oneview-sdk/resource/api500/c7000/storage_pool.rb#L59) that receives a Boolean, and in this method, it has a routine that raises an exception when it tries to unmanage a StoreVirtual that is not allowed. It might be interesting to use it instead of the ```update``` method.

Reply:
perfect